hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheCreateConfigOperation.java

```diff
@@ -90,7 +87,9 @@ public String getServiceName() {
     public void run() throws Exception {
         ICacheService service = getService();
         if (!ignoreLocal) {
-            response = service.putCacheConfigIfAbsent(config);
+            CacheConfig cacheConfig =
+                    config instanceof PreJoinCacheConfig ? ((PreJoinCacheConfig) config).asCacheConfig() : config;
```

Comment:
Prevent leaking the `PreJoinCacheConfig` to the cache service's known `CacheConfig`s, as it is only meant to be used for serialization/deserialization.

Reply:
Can we always do this check in `readInternal()`, or will this break the `if (createAlsoOnOthers)` part?

Reply:
Exactly, we want to continue using the `PreJoinCacheConfig` (if one was used to create this `CacheCreateConfigOperation`) when propagating the config to other members. We want it copied to a `CacheConfig` only when registered in `CacheService`.